### ISSUE TITLE
The supplied grant_type [urn:ietf:params:oauth:grant-type:jwt-bearer] is not supported.

### ISSUE THREAD

**Issue by cvbc2010** [AzureAD/azure-activedirectory-library-for-dotnet#1252]
**Which Version of ADAL are you using ?**
Microsoft.Identity.Client 2.0.0 Preview

**Which platform has the issue?**
.Net Core

**What authentication flow has the issue?**
* Desktop 
    * [ ] Interactive
    * [ ] WIA
    * [ ] U/P
    * [ ] Device code flow (browserless)
* [ ] Mobile
* Web App
    * [ ] Authorization code
    * [ ] OBO
* Web API
    * [x] OBO

Other? - please describe;

**Repro**

```ConfidentialClientApplication cca = new ConfidentialClientApplication(_azureAdB2c.ClientId, _azureAdB2c.Authority, _azureAdB2c.RedirectUri, new ClientCredential(_azureAdB
![msal obo issue](https://user-images.githubusercontent.com/13792108/45408139-1d1cc680-b689-11e8-99ff-3b52758d4a43.jpg)
2c.ClientSecret), null, null);
            try
            {
                AuthenticationResult result = await cca.AcquireTokenOnBehalfOfAsync(_azureAdB2c.ApiScopes.Split(' '),new UserAssertion(Request.Headers["Authorization"].ToString().Split(' ')[1]));
```

**Expected behavior**
It Should return another token for accessing another API.

**Actual behavior**
AADB2C90086: The supplied grant_type [urn:ietf:params:oauth:grant-type:jwt-bearer] is not supported.
Correlation ID: 0dd4da66-35da-4bce-b49d-6bc663efb343
Timestamp: 2018-09-12 06:58:23Z

**Possible Solution**
----

**Comment by henrik-me** [comment]
@cvbc2010 : based on the data you provided here it's a bit tricky to figure out what you are passing in. Can you pls. provide some additional data and a bit more details on the repro?


From the name of the scope variable it seems like you have a b2c app, you have to ensure your grant type is supported by B2C.

[B2C setup](https://docs.microsoft.com/en-us/azure/active-directory-b2c/active-directory-b2c-setup-aad-custom)
[B2C Documentation](https://docs.microsoft.com/en-us/azure/active-directory-b2c/)

**Comment by henrik-me** [comment]
@cvbc2010 : You are effectively using MSAL and not ADAL given the library you are using. Typically such questions should be posted here in our [MSAL repo](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues)

**Comment by bgavrilMS** [comment]
@cvbc2010 - we fixed some B2C issues in MSAL 2.0.1 - can you try with that? Please reopen the issue on the MSAL repo if you still have problems. 

https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues
